tests/foreman/cli/test_satellitesync.py

```diff
@@ -98,182 +94,141 @@ def docker_repo(module_target_sat, module_org):
+    def test_positive_export_library_custom_repo(
+        self, target_sat, export_import_cleanup_function, function_org, function_synced_repo
     ):
-        """Export custom repo via incremental library export
+        """Export custom repo via complete and incremental library export.
 
         :id: ba8dc7f3-55c2-4120-ac76-cc825ef0abb8
 
-        :expectedresults: Repository was successfully exported, exported files are
-            present on satellite machine
+        :setup:
+            1. Product with synced custom repository.
+
+        :steps:
+            1. Create a CV, add the product and publish it.
+            2. Export complete library.
+            3. Export incremental library.
+
+        :expectedresults:
+            1. Complete export succeeds, exported files are present on satellite machine.
+            2. Incremental export succeeds, exported files are present on satellite machine.
 
         :CaseLevel: System
         """
-        # Create custom product and repository
+        # Create cv and publish
         cv_name = gen_string('alpha')
-        product = make_product({'organization-id': function_org.id})
-        repo = make_repository(
-            {
-                'download-policy': 'immediate',
-                'organization-id': function_org.id,
-                'product-id': product['id'],
-            }
+        cv = target_sat.cli_factory.make_content_view(
+            {'name': cv_name, 'organization-id': function_org.id}
         )
-        # Create cv and publish
-        cv = make_content_view({'name': cv_name, 'organization-id': function_org.id})
-        ContentView.add_repository(
+        target_sat.cli.ContentView.add_repository(
             {
                 'id': cv['id'],
                 'organization-id': function_org.id,
-                'repository-id': repo['id'],
+                'repository-id': function_synced_repo['id'],
             }
         )
-        ContentView.publish({'id': cv['id']})
+        target_sat.cli.ContentView.publish({'id': cv['id']})
         # Verify export directory is empty
         assert target_sat.validate_pulp_filepath(function_org, PULP_EXPORT_DIR) == ''
-        # Export complete library, then export incremental
-        ContentExport.completeLibrary({'organization-id': function_org.id})
-        ContentExport.incrementalLibrary({'organization-id': function_org.id})
-        # Verify export directory is not empty
-        assert target_sat.validate_pulp_filepath(function_org, PULP_EXPORT_DIR) != ''
+        # Export complete and check the export directory
+        target_sat.cli.ContentExport.completeLibrary({'organization-id': function_org.id})
+        assert '1.0' in target_sat.validate_pulp_filepath(function_org, PULP_EXPORT_DIR)
+        # Export incremental and check the export directory
+        target_sat.cli.ContentExport.incrementalLibrary({'organization-id': function_org.id})
```

Comment:
I think we're testing the complete and incremental library export with same contents, so can we add some step to upload rpm or sync extra custom repo or something, before doing the incremental library export?

Reply:
We can, but it does not make much sense until we assert it was really done. And for this we need to extend the case by import section and assert it there, which is planned, but not in this PR. This PR is rather about cleanup.